### PR TITLE
feat(ci): add helpful error when check-generated-code gha workflow fails

### PR DESCRIPTION
**Pull request by jiridanek**
## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/code-quality.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ jobs:
           if [[ -z "$clean" ]]; then
               echo "Empty git status --porcelain: $clean"
           else
+              echo "::error::Please run the command from the previous step (Rerun all code generators...) and commit the result."
```
Comment (jstourac) [inline .github/workflows/code-quality.yaml:26]: Please, instead of telling to run command from previous step, maybe we should explicitly say what to run.

Comment (jiridanek) [reply .github/workflows/code-quality.yaml:26]: Ok, then, let me change this.

Comment (jiridanek) [reply .github/workflows/code-quality.yaml]: done

File: .github/workflows/code-quality.yaml [hunk 2]
```diff
@@ -15,14 +15,15 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Rerun all code generators we have
-        run: python3 ci/cached-builds/gen_gha_matrix_jobs.py
+        run: bash ci/generate_code.bash
```
Comment (jstourac) [inline .github/workflows/code-quality.yaml:18]: I don't like the `.bash` suffix :P , I can see we have it in one file already `./ci/cached-builds/gha_lvm_overlay.bash`... still, do we want to extend this policy?

File: ci/generate_code.bash [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+if ! python3 ci/cached-builds/gen_gha_matrix_jobs.py; then
+  # this returns nonzero when running outside of github; detect and print explanation
+  if [[ ${GITHUB_ACTIONS:-} == true ]]; then exit 1
```
Comment (jstourac) [inline ci/generate_code.bash:6]: It's a pitty that the script fails when non-run in github... I consider this solution as a workaround that will be fixed and won't be needed in the near future; we may miss other reasons why the script failed when run locally with this approach.

File: .github/workflows/code-quality.yaml (hunk 2 above)
Comment (jiridanek) [reply .github/workflows/code-quality.yaml:18]: it's only tested with bash; when I get the mac, I will also test it on zsh

File: ci/generate_code.bash (hunk 3 above)
Comment (jiridanek) [reply ci/generate_code.bash:6]: Yeah, it's not a good job on my part.

File: .github/workflows/code-quality.yaml (hunk 2 above)
Comment (jstourac) [reply .github/workflows/code-quality.yaml:18]: What does the file suffix have to do with the which interpreter is used?

File: ci/generate_code.bash (hunk 3 above)
Comment (jiridanek) [reply ci/generate_code.bash]: I was not too had to fix, did it rn

**Comment by jstourac** [comment]
Thank you, I like this much more now!

/lgtm

**Comment by jiridanek** [comment]
/override ci/prow/runtime-rocm-pytorch-ubi9-python-3-9-pr-image-mirror
/override ci/prow/amd-runtimes-ubi9-e2e-tests
/override ci/prow/images

one rocm run out of ephemeral storage, and tests hit the chicken-and-egg problem from https://github.com/openshift/release/pull/54567

**Pull request merged**